packages/@aws-cdk/aws-apigatewayv2/lib/http/stage.ts

```diff
@@ -163,20 +175,71 @@ export class HttpStage extends HttpStageBase {
       physicalName: props.stageName ? props.stageName : DEFAULT_STAGE_NAME,
     });
 
+    this.api = props.httpApi;
+    this.baseApi = props.httpApi;
+    this.stageName = this.physicalName;
+
+    const rateLimits = !props.throttle ? undefined : {
+      throttlingBurstLimit: !props.throttle.burstLimit ? undefined : props.throttle.burstLimit,
+      throttlingRateLimit: !props.throttle.rateLimit ? undefined : props.throttle.rateLimit,
+    };
+
+    const detailedMetrics = !props.detailedMetricsEnabled ? undefined : {
+      detailedMetricsEnabled: props.detailedMetricsEnabled,
+    };
+
+    let destinationArn: string | undefined = undefined;
+    if (props.accessLogEnabled) {
+      if (!props.accessLogGroupArn) {
+        // We need to set up the right permissions to create the log group.
+        const iamRoleForLogGroup = new Role(this, 'IAMRoleForAccessLog', {
+          roleName: PhysicalName.GENERATE_IF_NEEDED,
+          assumedBy: new ServicePrincipal('apigateway.amazonaws.com'),
+        });
+
+        iamRoleForLogGroup.node.addDependency(this.api);
+        iamRoleForLogGroup.applyRemovalPolicy(RemovalPolicy.DESTROY);
+        iamRoleForLogGroup.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('service-role/AmazonAPIGatewayPushToCloudWatchLogs'));
+
+        // It's required to register the iam role that is used to create the log group with the account
+        // if an api gateway has never been created in this account/region before.  Otherwise, this is
+        // a no-op.
+        const account = new CfnAccount(this, 'account', {
+          cloudWatchRoleArn: iamRoleForLogGroup.roleArn,
+        });
+        account.node.addDependency(this.api);
+
+        // Setting up some reasonable defaults for the retention policy and removal policy.
+        // If the user wants something different they should create their own log group.
+        const accessLogsLogGroup = new CfnLogGroup(this, 'AccessLoggingGroup', {
+          retentionInDays: 30,
+        });
+        accessLogsLogGroup.applyRemovalPolicy(RemovalPolicy.DESTROY);
```

Comment:
This should not be set to DESTROY.

Reply:
It seems to me that normally users are going to want the log groups to be deleted if the stack they're contained in gets deleted, which is why I chose to use DESTROY (and if they don't want that behavior they can supply a log group of their own with whatever behavior they prefer).  If that's not the convention what should I use?  RETAIN or just let it default (which I think is RETAIN)?

Reply:
yeah just leave the default (don't set anything)

Reply:
Done